frontend/src/pages/warehouse/ModalAddWarehouse.js

```diff
@@ -0,0 +1,367 @@
+    const filterByUsername = () => true;
+
+    const handleStateSearch = (query) => {
+        fetch(`/customers/${props.currentCustomerId}/states?state=${query}`)
+            .then(resp => resp.json())
+            .then(res => {
+                setStateOptions(res);
+            });
+    };
+
+    const handleDispatcherSearch = (query) => {
+        fetch(`/customers/${props.currentCustomerId}/users/dispatchers?username=${query}`)
+            .then(resp => resp.json())
+            .then(res => {
+                const optionsFromBack = res.map((dispatcher) => ({
+                    id: dispatcher.id,
+                    name: dispatcher.name,
+                    surname: dispatcher.surname,
+                    username: dispatcher.username
+                }));
+                setDispatcherOptions(optionsFromBack);
+            });
+    };
+
+    const onChangeState = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {
+                ...preState.addressDto,
+                state: (e.length ?
+                    {id: e[0].id, state: e[0].state}
+                    : {id: '', state: ''})
+            }
+        }));
+    };
+
+    const addDispatcher = (e) => {
+        if (e.length) {
+            const isContains = dispatchers.find(disp => e[0].id === disp.id)
+            if (!isContains) {
+                e.map(dispatcher =>
+                    setDispatchers(preState => ([
+                        ...preState, dispatcher
+                    ])));
+            }
+        }
+    };
+
+    const handleIdentifier = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            identifier: e.target.value
+        }));
+    };
+
+    const handleType = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            type: e
+        }));
+        setDropdownMenuName(e);
+    };
+
+    const handleCity = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, city: e.target.value}
+        }));
+    };
+
+    const handleLineOne = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, addressLine1: e.target.value}
+        }));
+    };
+
+    const handleLineTwo = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, addressLine2: e.target.value}
+        }));
+    };
+
+    const handleTotalCapacity = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            totalCapacity: e.target.value
+        }));
+    };
+
+    const addWarehouseHandler = (e) => {
+        e.preventDefault();
+
+        let updateWarehouseDto = {};
+        if (dispatchers.length) {
+            const dispatchersId = dispatchers.map(dispatcher => dispatcher.id);
+            updateWarehouseDto = {
```

Comment:
updateWarehouseDto sounds like operation, but it's the name of the variable. warehouseUpdateDto probably will be more understandable.